### PR TITLE
*: clean code for restore table

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
change 
```sql
admin restore table [table_name]
``` 
syntax to 
```sql
recover table [table_name]
```
syntax.
clean code for restore table.
related PR: https://github.com/pingcap/tidb/pull/7937
related parser PR: https://github.com/pingcap/parser/pull/180

### What is changed and how it works?
move restore table from adminStmt to ddl stmt. Because recover table is a ddl operation. 
remove redundancy code.

### Check List 

Tests 

 - No test

Code changes

 - Has exported function/method change

Side effects

Related changes

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Please fix ci @crazycs520

**Comment by tiancaiamao** [comment]
LGTM
Is there a document about this statement and should we also update it?

**Comment by crazycs520** [comment]
The [proposal](https://github.com/pingcap/tidb/blob/master/docs/design/2018-08-10-restore-dropped-table.md) of this is use the `ADMIN RESTORE TABLE table_id` syntax, should we update the proposal? @winkyao

**Comment by winkyao** [comment]
@crazycs520 I don't think it is necessary, but you can add a paragraph to explain the implement is not the same with the proposal.

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -285,8 +285,8 @@ func (w *worker) finishDDLJob(t *meta.Meta, job *model.Job) (err error) {
 		}
 	}
 	switch job.Type {
-	case model.ActionRestoreTable:
-		err = finishRestoreTable(w, t, job)
+	case model.ActionRecoverTable:
```
Comment (winkyao) [inline ddl/ddl_worker.go]: What about the compatibility, if the job is pushed by old version tidb, the model.ActionRestoreTable job type will be treated as invalid job?

**Comment by winkyao** [comment]
Please resolve the conflicts

**Pull request merged**